src/components/buttons/icon-button/icon-button.js

```diff
@@ -1,26 +1,25 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
 import { compose } from 'recompose';
-import invariant from 'tiny-invariant';
 import isNil from 'lodash.isnil';
+import { css } from '@emotion/core';
+import vars from '../../../../materials/custom-properties';
 import withMouseDownState from '../../../hocs/with-mouse-down-state';
 import withMouseOverState from '../../../hocs/with-mouse-over-state';
 import filterDataAttributes from '../../../utils/filter-data-attributes';
 import AccessibleButton from '../accessible-button';
-import styles from './icon-button.mod.css';
-
-const getStateClassNames = (isDisabled, isToggled) =>
-  classnames({
-    [styles.default]: !isToggled && !isDisabled,
-    [styles.active]: isToggled,
-    [styles.disabled]: isDisabled,
-  });
+import {
+  getStateStyles,
+  getShapeStyles,
+  getSizeStyles,
+  getThemeStyles,
+} from './icon-button.styles';
 
 // Gets the color which the icon should have based on context of button's state/cursor behavior
 const getIconThemeColor = props => {
+  const isActive = props.isToggleButton && props.isToggled;
```

Comment:
Based on the prop types validation, the active state is only set when both props are true.

Please correct me if I'm wrong...

Reply:
I us previously fiddling a lot with this but it looks still good.